xbmc/cores/playercorefactory/PlayerCoreConfig.h

```diff
@@ -104,16 +99,19 @@ class CPlayerCoreConfig
     else
       return nullptr;
 
+    if (!pPlayer)
+      return nullptr;
```

Comment:
If you want to go further, `CPlayerCoreConfig::CreatePlayer()` should return a unique_ptr

Reply:
I've actually done this already and will prepare a PR soon 😄